File: app/src/main/java/com/tomaszrykala/dorefindmi/things/supplier/ledstrip/LedStripSupplier.java
package com.tomaszrykala.dorefindmi.things.supplier.ledstrip;

import com.tomaszrykala.dorefindmi.model.Led;
import com.tomaszrykala.dorefindmi.things.supplier.BaseSupplier;

public interface LedStripSupplier extends BaseSupplier {

    boolean isLitAt(Led led);

    void light(Led led);

    void reset();
}

File: app/src/main/java/com/tomaszrykala/dorefindmi/things/supplier/ledstrip/LedStripSupplier.kt
package com.tomaszrykala.dorefindmi.things.supplier.ledstrip

import com.tomaszrykala.dorefindmi.things.supplier.BaseSupplier

const val TURN_OFF_COLOR = -1

interface LedStripSupplier : BaseSupplier {

    fun light(colors: IntArray)

    fun getLength(): Int
}
